x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/cache/FrozenCacheService.java

```diff
@@ -327,8 +330,8 @@ public CacheFileRegion get(CacheKey cacheKey, long fileLength, int region) {
     }
 
     public void onClose(CacheFileRegion chunk) {
-        assert regionOwners[chunk.sharedBytesPos].compareAndSet(chunk, null);
-        freeRegions.add(chunk.sharedBytesPos);
+        assert regionOwners[chunk.sharedPageIndex].compareAndSet(chunk, null);
+        freeRegions.add(chunk.sharedPageIndex);
```

Comment:
Maybe also `assert chunk.isEvicted();`, just in case

Reply:
++